docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+
+The two file formats should be handled automatically in `SourceModel.read()` and
+`SourceModel.write()`.
+
+
+Improve spatial models
+----------------------
+
+Implement sky coordinate handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Implement support for coordinate systems. Add an interface for `SkyCoord`, by
+introducing a `SpatialModel.skycood` attribute.
+
+.. code-block::
+
+    from astropy.coordinates import SkyCoord
+    from gammapy.image.models import PointSource
+
+    # instanciation using a SkyCoord object
+    point_source = PointSource(skycood=SkyCoord(0, 0, frame="galactic", unit="deg"))
+    # or
+    point_source = PointSource(lon="0 deg", lat="0 deg", frame="galactic")
+
+    skycoord = point_source.skycoord
+    assert isinstance(skycoord, SkyCoord)
+
+    # evaluation of models using sky coordinates
+    coords = SkyCoord([0, 1], [0.5, 0.5], frame="galactic", unit="deg")
+    values = point_source(coords)
+
+    # maybe override evaluation for efficient evaluation for fitting
+    values = point_source(lon, lat)
+
+
+Implement default parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For spectral models Gammapy has defined default parameters, which simplifes the
+interactive usage of the models. From this experience we propose to introduce
+default parameters for `SpatialModel` as well as `SourceModel`.
+
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+    from gammapy.cube.models import SourceModels
+
+    shell = Shell()
+    print(shell)
+
+
+Implement evaluation region specifications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For efficient evaluation of model components, we propose that `SpatialModel`
+classes declare a circular region `.evaluation_radius`, they are evaluated on.
+
+.. code-block::
+
+    from gammapy.image.models import Shell
+
+    shell = Shell()
+    print(shell.evaluation_radius)
+
+The `.evaluation_radius` should be a property of the model, that is computed
+depending of the model parameter, that determines the spatial extension of the
+model. The handling of `PointSource` and additional evaluation margins, which depend
+on the pixel size, is done in the step, where the model component is evaluated.
+
+
+Expose model parameters as attributes
+-------------------------------------
+To simplify the interactive handling of model instances, model parameters should
+be exposed as attribues on the model, so that the following works:
+
+.. code-block::
+
+    from gammapy.spectrum.models import Powerlaw
+
+    pwl = Powerlaw()
+
+    pwl.amplitude.value = "1e-12 cm-2 s-1 TeV-1"
+
+    # instead of
+
+    pwl.parameters["amplitude"].value = "1e-12 cm-2 s-1 TeV-1"
+
+The attributes should be auto-genrated in the `Model` base class, based on the
+list of parameters defined by the model.
+
+Introduce sub-module for models
+-------------------------------
+
+We propose to introduce a `gammapy.models` or `gammapy.modeling` sub-module, which
+bundles all the model classes in single namespace.
```

Comment:
Do you mean to only expose models in `gammapy.models`?
Or to also expose them where they are now, and to expose them via a second namespace in addition?
Please clarify?

Also, I think there should be a minimal answer of 1-2 sentences on "why"? This is a major decision concerning code organisation in Gammapy. It might be best to just defer this question to a later PIG and simply leave as-is for now?
(and not discuss / decide whether to introduce `gammapy.models`) since it's not needed to build the features?

Like I said earlier: I assume that even from Python scripts, the most common way to create models will be to call a factory function, passing a "model spec" string or dict as input, making the question of model class names and namespace locations less important.

Reply:
I removed the suggestion to introduce a separate name space. It will be discussed in the context of the whole sub-package structure for Gammapy in separate PIG, before Gammapy v1.0